### PR TITLE
Use server load/save endpoints from front end

### PR DESCRIPTION
**Pull request by reddigari**
- Load/save on front-end hits endpoints @reelmatt  set up #27
     - Save sends the diagram data, receives the response containing the diagram and backend serializations, and begins a download
     - Load prompts user for file upload, sends to server, receives response, and deserializes/renders diagram
- Workspace occupies full width of screen

### CODE REVIEW

File: front-end/src/components/Workspace.js [hunk 1]
```diff
@@ -35,16 +35,27 @@ class Workspace extends React.Component {
...
-        anchor.href = dataStr;
-        anchor.download = "diagram.json";
-        anchor.click();
-        anchor.remove();
+        const resp = await fetch("/workflow/save", {
+            method: "POST",
+            headers: {"Content-Type": "application/json"},
+            body: data
+        });
+        if (resp.status !== 200) {
```
Comment (reelmatt) [inline front-end/src/components/Workspace.js]: You should add this check for 200 in the `FileUpload` function below as well.

**Comment by reddigari** [comment]
@reelmatt Hmmm I can't seem to replicate it. Can you take a screen recording?

**Comment by reelmatt** [comment]
Slightly different behavior then what I was experiencing earlier. I can get it to clear and load (unlike before), but a second clear seems to cause a problem? My thinking was it was whatever 'clear' was doing, but that doesn't seem to be the case.

[Screen recording](https://www.dropbox.com/s/otfvniq6021b49q/Clear-load%20bug.mov?dl=0)

**Comment by reddigari** [comment]
Excellent catch. The event listener is for `onchange`, so if you select the same file, the listener doesn't fire. I'll fix it to reset the file between each event.

File: front-end/src/components/Workspace.js (hunk 1 above)
Comment (reddigari) [reply front-end/src/components/Workspace.js]: https://github.com/matthew-t-smith/visual-programming/pull/28/commits/b2404dd359d1d399613f9624d67ccf27492f5712

**Pull request merged**